src/paho/mqtt/client.py

```diff
@@ -1058,7 +1089,7 @@ def username_pw_set(self, username, password=None):
         username: The username to authenticate with. Need have no relationship to the client id.
         password: The password to authenticate with. Optional, set to None if not required.
         """
-        self._username = username.encode('utf-8')
```

Comment:
Yes, because in fixes the .encoding was moved to _send_connect and in develop the encoding of password was added.
So it's the merge that caused encoding to be done in two place.